service-coordtransform/src/main/java/fi/nls/paikkatietoikkuna/coordtransform/CoordinateTransformationActionHandler.java

```diff
@@ -154,16 +433,17 @@ private String getTargetCrs(ActionParameters params) throws ActionParamsExceptio
 
     private void assertNumber(JsonToken token, String err) throws ActionParamsException {
         if (token != JsonToken.VALUE_NUMBER_FLOAT && token != JsonToken.VALUE_NUMBER_INT) {
-            throw new ActionParamsException(err);
+            throw new ActionParamsException(err, "invalid_number");
         }
     }
 
-    protected void writeJsonResponse(OutputStream out, List<Coordinate> coords, final int dimension)
+    protected void writeJsonResponse(OutputStream out, List<Coordinate> coords, List<Coordinate> inputCoords, final int dimension)
             throws ActionException {
         try (JsonGenerator json = jf.createGenerator(out)) {
             json.writeStartObject();
-            json.writeNumberField("dimension", dimension);
-            json.writeFieldName("coordinates");
+            json.writeNumberField(RESPONSE_DIMENSION, dimension);
+            json.writeBooleanField("hasMoreCoordinates", this.hasMoreCoordinates);
```

Comment:
Okay so the hasMoreCoordinates flag is returned to the client indicating that "some of the coordinates you sent use were ignored". However there's no way to continue the process. What's the use case?

Reply:
If user is transforming large file to array/table then it will be noticed that "you have to transform to file to get all results". Table shows only 100 first coordinates to get a preview. Splitting large files to smaller queries or using POST isn't implemented yet.

Reply:
OK